packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -342,7 +237,53 @@ export class SchemaFactoryRecursive<
 					[string, InsertableTreeNodeFromImplicitAllowedTypesUnsafe<T>]
 				>;
 			},
-			false
+			false,
+			T
 		>;
 	}
 }
+
+/**
+ * Compile time check for validity of a recursive schema.
+ *
+ * @example
+ * ```typescript
+ * 	class Test extends sf.arrayRecursive("Test", [() => Test]) {}
+ *	{
+ *		type _check = ValidateRecursiveSchema<typeof Test>;
+ *	}
+ * ```
+ * @remarks
+ * The type of a recursive schema can be passed to this, and a compile error will be produced for some of the cases which the schema in malformed.
+ * This can be used to help mitigate the issue that recursive schema definitions are {@link Unenforced}.
+ * If an issue is encountered where a mistake in a recursive schema is made which produces an invalid schema but is not rejected by this checker,
+ * it should be considered a bug and updated to handle that case (or have a disclaimer added to these docs that it misses that case).
+ * @privateRemarks
+ * There are probably mistakes this misses: its hard to guess all the wrong things people will accidentally do and defend against them.
+ * Hopefully over time this can grow toward being robust, at least for common mistakes.
+ * @beta
+ */
+export type ValidateRecursiveSchema<
+	T extends TreeNodeSchemaClass<
```

Comment:
Any inline docs to explain the components would be super helpful for anyone wanting to modify this in the future.